### ISSUE TITLE
download plugin can't upload bigger file

### ISSUE THREAD

**Issue by Jimmi08** [e107inc/e107#1809]
I have file cca 69MB.
My setting for upload is 180MB
![image](https://cloud.githubusercontent.com/assets/5429548/17832820/c209cbc6-670c-11e6-9f99-29294d2002de.png)

My php.ini is 128MB
![image](https://cloud.githubusercontent.com/assets/5429548/17832822/d56b0fcc-670c-11e6-8820-c2ca9247703a.png)
![image](https://cloud.githubusercontent.com/assets/5429548/17832824/d9371470-670c-11e6-9b70-24ac2de7ba8c.png)
![image](https://cloud.githubusercontent.com/assets/5429548/17832825/dd3e4f16-670c-11e6-9cb2-00fd31124369.png)

But I get this error:
![image](https://cloud.githubusercontent.com/assets/5429548/17832826/fb53c9fe-670c-11e6-8aed-c96f5d13577c.png)

Uploading is working, I had no problem to upload 20MB file.  Thanks

**Comment by tgtje** [comment]
Just test @Jimmi08 i have ( **local**) max filesize on prefs  > empty > let php ini take it.
Settings in php ini accordingly ( by size ).

Go to . . /e107_web/js/core > mediaManager (js)  about line 533  > max_file_size : "20mb"  
Change and try again...  Keep in mind  time !!! 
(have tested with up to 182MB zip; and 113MB mp3)
Both with success screen of mp3 : 
![yow](https://cloud.githubusercontent.com/assets/1811857/17836559/e3cefca6-6796-11e6-9240-cf640e90abfc.jpg)

**Comment by Jimmi08** [comment]
Thank you for help.
I have max filesize on prefs empty
![image](https://cloud.githubusercontent.com/assets/5429548/17836727/d6ddbdce-679c-11e6-9b11-412f242291e7.png)

I set php.ini and filexml to 128MB
![image](https://cloud.githubusercontent.com/assets/5429548/17836730/f9b893b4-679c-11e6-87ac-ae651b285b44.png)

It worked on localhost.

But not on live site.  I've double check that I upload my changes.

**Comment by Moc** [comment]
@Jimmi08 Maybe a caching issue on the live version?

**Comment by Jimmi08** [comment]
Uff, you were right. I delete cache (not enabled, but for sure) in e107 admin and it worked. THANKS TO ALL.   But this is still bug, I will lose this change with next update, won't I?

**Comment by Moc** [comment]
Yep labelled as bug. Waiting for a review from Cameron to ensure this is the right fix, then he or I will commit it. 

Thanks for reporting :+1:

**Comment by tgtje** [comment]
Need to add info ( for readers), that this (pref setting for php ini) only is applicable for hosts that do allow such.. ( mine not, so i tested locally).

**Comment by Jimmi08** [comment]
I need to customize next parameter, file extension, it's ignored filetype.xml
![image](https://cloud.githubusercontent.com/assets/5429548/17907911/4d17478e-697e-11e6-84b4-38daff25bacd.png)

![image](https://cloud.githubusercontent.com/assets/5429548/17907928/598a5e48-697e-11e6-8ffc-907215f7ac1f.png)

**Comment by tgtje** [comment]
Thats a good pointer @Jimmi08  :+1:   So basically theres filetypes (allowed or not) and a filter that overlaps, but are NOT identical. two methods that CAN create issues when certain types are being used.
While (found perhaps a new bug on upload itself), plupload js IN combination with filet.xml works (i can upload a m4a) ; **this** issue relates to mediaManager upload/filter.

**Comment by Moc** [comment]
Summary:
- Filesize and filetypes are hardcoded in  /e107_web/js/core/mediaManager.js

**Comment by chimcen** [comment]
Stupid question - but could this be a possible reason:?
In mediaManager.js Line 555 ends the filetypes list with a comma (other lines with filetypes end without)...

`{title : "Image files", extensions : extImg || "jpg,gif,png,jpeg"},
 {title : "Zip files", extensions : extArchive || "zip,gz,rar"},
 {title : "Document files", extensions : extDoc || "pdf,doc,docx,xls,xlsm,xml"},
 {title : "Media files", extensions: 'mp3,mp4,wav,ogg,webm,mid,midi,'},
 {title : "Other files", extensions: 'torrent,txt'}`

Another thing - it would be great when adding tiff, 7z and avi to defaults...

**Comment by Moc** [comment]
The issue here is that these values (size and file type) are hard coded in a place where they shouldn't be. They should be customisable through the filetypes.xml file, in which you can also add or change current filetypes and sizes.

**Comment by rica-carv** [comment]
Or why not use a preferences field for it?
That way, all could be donne via admin interface area...

**Comment by Moc** [comment]
Yeah there was some issue with this in the past. Not sure what it was, maybe security reasons, and then it was decided to use a physical file rather than prefs. Anyhow, I'll leave this for someone else to answer.

**Comment by CaMer0n** [comment]
I don't think plupload supports different file-sizes by type. Recent changes I made should allow uploading via media-manager with the limit stored in the pref.  We may need to allow ALL file-types through the media-manager.

**Comment by Jimmi08** [comment]
to have it in one place (it's something different, I know, but this will be solved, just think on other ideas)
https://github.com/e107inc/e107/issues/3169

**Comment by CaMer0n** [comment]
It's currently at 400mb.
